social-network-data-migration-core/src/main/java/pl/edu/agh/socialnetworkdatamigration/core/migrator/Migrator.java

```diff
@@ -1,7 +1,69 @@
 package pl.edu.agh.socialnetworkdatamigration.core.migrator;
 
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.Record;
+import org.neo4j.driver.Result;
+import org.neo4j.driver.Session;
 import pl.edu.agh.socialnetworkdatamigration.core.mapping.SchemaMapping;
 
-public interface Migrator<S extends SchemaMapping> extends AutoCloseable {
-    void migrateData(S schemaMapping);
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class Migrator<S extends SchemaMapping> implements AutoCloseable {
+    private final ExecutorService executor = Executors.newCachedThreadPool();
+    private final MigrationStrategy<S> migrationStrategy;
+    private final Driver neo4jDriver;
+    private final boolean dryRun;
+
+    public Migrator(Driver neo4jDriver, MigrationStrategy<S> migrationStrategy, boolean dryRun) {
+        this.migrationStrategy = migrationStrategy;
+        this.neo4jDriver = neo4jDriver;
+        this.dryRun = dryRun;
+    }
+
+    public void migrateData(S schemaMapping) {
+        List<Set<String>> operations = migrationStrategy.createMigrationQueries(schemaMapping);
+        operations.forEach(this::waitForQueries);
+    }
+
```

Comment:
Migrator w takiej formie, z własnym `ThreadPool` powinien być raczej tylko jeden. W związku z tym zmieniłbym nazwę klasy z `Migrator<S extends SchemaMapping>` na po prostu `Migrator`. 

Tę definicję typu generycznego S można przenieść do metody, chyba w taki sposób:
`public <S extends SchemaMapping> void migrateData(S schemaMapping)`

`MigrationStrategy<S> migrationStrategy` i `dryRun` powinny być moim zdaniem parametrem nie konstruktora, a `migrateData`.

Potem w serwerze w Springu mógłbym po prostu stworzyć taki jeden migrator i go wstrzykiwać zamiast tworzyć nowego do każdej osobnej migracji

Zastanawia mnie, czy neo4jDriver też nie warto przenieść do `migrateData` - to pewnie zależy, czy w serwerze chcemy mieć możliwość migrowania do wielu baz neo4j (wtedy neo4jDriver powinien być parametrem migrateData) czy zakładamy się podpinamy się do jednej (jak teraz)